core/socket_data.py

```diff
@@ -89,7 +89,7 @@ def SvGetSocket(socket, deepcopy=True):
     set to False and increase performance substanstilly
     """
     global socket_data_cache
-    if socket.is_linked:
+    if socket.links:
```

Comment:
i recall from @ly29 that there was a good reason to favour using ".is_linked" instead of ".links"

https://github.com/nortikin/sverchok/issues/454

`.links` may come with a cpu penality,  the decision to use it was not arbitrary.

Reply:
but hey, we just want stuff to work. so maybe we could track our own `.sv_links` ...eventually.  or propose an improvement to the pynode api.

Reply:
.links is actually slow enough to be noticed from time to time. The blow is the implemetation.

```
@property
def links(self):
    """List of node links from or to this socket. Warning: takes O(len(nodetree.links)) time."""
    return tuple(
        link for link in self.id_data.links
        if (link.from_socket == self or
            link.to_socket == self))
```

Reply:
I am also quite pleased that my old code is still used.

Reply:
according this #2772,solution should not make performance more lousy than it is now.))) 

I agree of course that it is  much more slower. But I think it is better to have more robust solution then faster. I agree that `sv_links` can solve the problem but I think in independent PR. 
`if self.links:` can be temporary solution.

Reply:
What is the problem with .is_linked in this context?

Reply:
The problem is that when one of a sockets of `dictionary in` node is being plugged it cause update event during which the attribute `is_linked` of the socket returns False.

`Socket: Dict in sock, is_linked: False, links: True`

The result is that during process of the node `sv_get` can't get catched data from the socket.

This is only one case when `is_linked` is not appropriate, in all other cases it works as expected.

Reply:
Well, the reason why we did not faced with such problem before is that in `dictionary in` node I replace input sockets by another socket with type according type of `other` socket. 

I don't know probably I should not do this but it looks like that node tree looks more readable with type of inputs socket equal to type of socket to which their are linked.

Reply:
If variant below is acceptable then probably we could avoid using `socket.links` test.

![2019-12-29_23-51-37](https://user-images.githubusercontent.com/28003269/71561889-2e642f00-2a96-11ea-94dc-e69bdd93d023.png)

Reply:
Probably possible variant is to create `sv_is_linked` method for sockets which will be False by default but can be changed manually. And then such checking should work `if (socket.is_linked or socket.sv_is_linked) or something else:`